File: Customer Account Tracker FrontEnd/customeraccounttrackerapp/src/services/TransactionService.js
import axios from 'axios';

const TRANSACTION_BASE_URL = "http://localhost:8081/user"

class TransactionService {


    sendMoney(senderAccNum, receiverAccNum, amount) {
        return axios.post(`${TRANSACTION_BASE_URL}/${senderAccNum}/send-money/${receiverAccNum}?amount=${amount}`)
    }
    withdrawMoney(accNum, amount) {
        return axios.post(`${TRANSACTION_BASE_URL}/${accNum}/withdraw-money?amount=${amount}`)
    }
    depositMoney(accNum, amount) {
        return axios.post(`${TRANSACTION_BASE_URL}/${accNum}/deposit-money?amount=${amount}`)
    }

    getTransactionHistory(accNum){
        return axios.get(`${TRANSACTION_BASE_URL}/${accNum}/transactions`)
    }

    deleteTransaction(id){
        return axios.delete(`${TRANSACTION_BASE_URL}/delete-transaction/${id}`)
    }

}

export default new TransactionService();

File: Backend/customer_account_tracker/src/main/java/com/cts/customer_account_tracker/Service/TransactionService.java
package com.cts.customer_account_tracker.Service;

import java.util.List;

import com.cts.customer_account_tracker.Entity.BankAccount;
import com.cts.customer_account_tracker.Entity.Transaction;

public interface TransactionService {

	public void sendMoney(String senderAccNum, String receiverAccNum, double amount);
	public void withdrawMoney(String userAccNum, double amount);
	public void depositMoney(String userAccNum, double amount);
	public Transaction createTransaction(BankAccount acc, String transactionType, double amount);
	public List<Transaction> getTransactionHistory(String userAccNum);
	public String deleteTransactions(Long id);
}
